crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -815,6 +815,12 @@ pub struct RunArgs {
     #[clap(long)]
     pub no_cache: bool,
 
+    /// If outputs have been specified, but none exist - skip saving the (empty)
+    /// cache for the task. Enabling this can be a useful safety measure to
+    /// ensure proper outputs configuration.
+    #[clap(long, value_name = "BOOL", action = ArgAction::Set, default_value = "false", default_missing_value = "true", num_args = 0..=1)]
+    pub skip_empty_cache: bool,
```

Comment:
Not sold on adding this as an arg (or this name if we do), but the behavior is nice to have - Enabling this fixes the case when you set outputs incorrectly, and nothing ends up being cached. This can be problematic because if you don't know this happened, and you don't make changes you can get a cache hit that also nothing, making it look like everything worked.

Reply:
I'm not super for just adding the flag. This is a setting that is should generally be set at the repo level. e.g. If CI has `--skip-empty-cache` a user can poison the cache with a bad run if they don't have it set.